Improve documentation for #addin preprocessor directive (#1768)
cake-build/website: input/docs/writing-builds/preprocessor-directives/addin.md

Order: 10
Title: Addin directive
Description: Directive to install and reference addins using NuGet
---

Supported in: **[Cake .NET Tool] | [Cake runner for .NET Framework] | [Cake runner for .NET Core]**

The addin directive is used to install and reference addins using NuGet.

:::{.alert .alert-info}

> Addin directives is not available when running with [Cake Frosting].

Reviewer: ```suggestion
The `#addin` directive is not available when running with [Cake Frosting].
```

or

```suggestion
Addin directives are not available when running with [Cake Frosting].
```

Instead addins can be added as NuGet packages to the [Cake Frosting] project.
:::

> 

Reviewer: ```suggestion
#addin nuget:?package=Cake.Foo&version=1.0.0
```

Perhaps start with the most common example of package ID and version before expanding on details?

The following URI parameters are supported by the addin directive.

# Source

This is not a named parameter, but it is permitted as per the URI definition.
By default, the provider will attempt to install addins from nuget.org.
If the package is hosted on another feed the installation source can be overridden.

## Example

Download from myget.org:

```csharp
#addin nuget:https://myget.org/f/Cake/?package=Cake.Foo
```

Install addin from a local directory:

```csharp

> #addin nuget:file://localhost/packages/?package=Cake.Foo

Reviewer: ```suggestion
#addin nuget:file:///C:/MyPackages/?package=Cake.Foo
```

Perhaps a more "Windows" example makes it easier to understand?

If not provided, the latest version package that is available will be installed.

## Example

```csharp
#addin nuget:?package=Cake.Foo&version=1.2.3
```

# Prerelease

To install prerelease tools without defining a specific `version` (`#addin nuget:?package=Cake.Foo&version=1.2.3-beta`), the `prerelease` parameter needs to be passed.

## Example

```csharp
#addin nuget:?package=Cake.Foo&prerelease
```

# Dependencies

:::{.alert .alert-success}
Available since Cake 0.22.0.
:::

By default dependencies defined on the NuGet package are not loaded.
The `loaddependencies` parameter allows to fetch and load dependent NuGet packages.

:::{.alert .alert-warning}
This feature requires Cake to be [configured to use in-process NuGet client] instead of external `nuget.exe`.
:::

## Example

```csharp
#addin nuget:?package=Cake.Foo&loaddependencies=true
```

# Include

The `include` parameter allows to define the files which should be included.

## Example

```csharp
#addin nuget:?package=Cake.Foo&include=/**/NoFoo.dll
```

# Exclude

The `exclude` parameter allows exclude specific files.

## Example

```csharp
#tool nuget:?package=Cake.Foo&exclude=/**/Foo.dll
```

> 

Reviewer: Is it worth mentioning somewhere that quotes can be used and are optional?

```cake
#addin nuget:?package=Cake.Foo
```
vs

```cake
#addin "nuget:?package=Cake.Foo"
```
